src/typescript/index.d.ts

```diff
@@ -0,0 +1,430 @@
+         */
+        gasPrice?: number | string | BN;
+        /**
+         * default value 3000000
+         */
+        readonly gasLimit?: number;
+        readonly to?: string;
+        readonly data: string;
+        /**
+         * Defaults to `"restricted"`
+         *
+         * Hyperledger Besu only implements `"restricted"` mode of operation.
+         */
+        readonly restriction?: "restricted" | "unrestricted";
+    }
+
+    export interface ISendRawTransactionOptions extends ISendGenericTransactionOptions, TransactionConfig {
+        /**
+         * Private Key used to sign transaction with
+         */
+        readonly privateKey: string;
+        /**
+         * Enclave public key
+         */
+        readonly privateFrom: string;
+        /**
+         * Enclave keys to send the transaction to
+         */
+        readonly privateFor: string[];
+        /**
+         * Enclave id representing the receivers of the transaction
+         */
+        readonly privacyGroupId?: string;
+        /**
+         * (Optional) : If not provided, will be calculated using`eea_getTransctionCount`
+         */
+        readonly nonce?: number;
+        /**
+         * The address to send the transaction
+         */
+        readonly to?: string;
+        /**
+         * Data to be sent in the transaction
+         */
+        readonly data: string;
+    }
+
+    export interface ISetPrivacyGroupLockState {
+        /**
+         * Privacy group ID to lock / unlock
+         */
+        readonly privacyGroupId: string;
+        /**
+         * Private Key used to sign transaction with
+         */
+        readonly privateKey: string;
+        /**
+         * Orion public key
+         */
+        readonly enclaveKey: string;
+        /**
+         * boolean indicating whether to lock or unlock
+         */
+        readonly lock: boolean;
+    }
+
+    export interface IAddToPrivacyGroupOptions {
+        /**
+         * Privacy group ID to add to
+         */
+        readonly privacyGroupId: string;
+        /**
+         * Private Key used to sign transaction with
+         */
+        readonly privateKey: string;
+        /**
+         * Orion public key
+         */
+        readonly enclaveKey: string;
+        /**
+         * list of enclaveKey to pass to the contract to add to the group
+         */
+        readonly participants: string[];
+    }
+
+    export interface IRemoveFromPrivacyGroupOptions {
+        /**
+         * Privacy group ID to add to
+         */
+        readonly privacyGroupId: string;
+        /**
+         * Private Key used to sign transaction with
+         */
+        readonly privateKey: string;
+        /**
+         * Orion public key
+         */
+        readonly enclaveKey: string;
+        /**
+         * single enclaveKey to pass to the contract to add to the group
+         */
+        readonly participant: string;
+    }
+
+    export interface ICreatePrivacyGroupOptions {
```

Comment:
In these 4 interfaces `ICreatePrivacyGroupOptions`, `IRemoveFromPrivacyGroupOptions`, `IAddToPrivacyGroupOptions` and `ISetPrivacyGroupLockState` have the fields `privacyGroupId`, `privateKey` and `enclaveKey`. Maybe create an `IPrivacyGroupOptions` that will have those fields and then the interfaces will extend it.

Reply:
@achraf17 Done: 0cd5cd64d78e230154d2be1c15d6c7daf5f8c7c2